docs/contributing/rdg-storage-format.rst

```diff
@@ -0,0 +1,71 @@
+=========================
+Katana RDG Storage Format
+=========================
+
+Katana persists its graphs on disk in a format called `RDG: Resilient Distributed Graph`. This `RDG` is a directory filled with files that define the graph.
+The `storage_format_version` of an RDG determines the layout of the RDG
+An example of an RDG can be found here
+https://github.com/KatanaGraph/test-datasets/tree/master/rdg_datasets/ldbc_003/storage_format_version_3
+
+Katana software uses the `storage_format_version` of a loaded RDG to determine some of the features it supports
+
+Katana software always stores the RDG as the latest `storage_format_version`, sometimes with the `unstable_storage_format` modifier (covered below)
+RDGs stored in a newer `storage_format_versions` are incompatible with older software.
+All versions of Katana after a `storage_format_version` is introduced should support loading/reading RDGs stores as that `storage_format_version`.
+
+Early RDGs from before `storage_format_version` was introduced are assumed to be `storage_format_version=1`
+
+Rarely, `storage_format_versions` can be deprecated. When this occurs tooling will be provided to convert all RDGs stored in that `storage_format_version` to the next supported `storage_format_version`.
+
+Unstable RDG Storage Format
+===========================
+
+The unstable RDG storage format lets developers work on and test persisting new data structures without disturbing everyone else that relies on the RDG format to be stable.
+The unstable RDG storage format should be used while developing features for Katana software which alter, add, or remove, data structures in the RDG format.
+
+Katana software executed *with* the `KATANA_ENABLE_EXPERIMENTAL="UnstableRDGStorageFormat"` environment variable supports loading/storing RDGs in the unstable RDG storage format.
+Katana software executed *without* the `KATANA_ENABLE_EXPERIMENTAL="UnstableRDGStorageFormat"` environment variable *does not* support loading/storing RDGs in the unstable RDG storage format.
+
+Katana software uses the boolean flag `unstable_storage_format` in the RDGs part header to identify unstable format RDGs
+
+If Katana software is executed *with* the `KATANA_ENABLE_EXPERIMENTAL="UnstableRDGStorageFormat"` environment variable:
+#. katana software stores RDGs:
+ 
+   #. with `storage_format_version = latest_storage_format_version`
+   #. with `unstable_storage_format = true`
```

Comment:
By making it a flag, it is much more obvious to developers, the software, and tools, that this RDG is in an unstable format. Usually making things more obvious results in clearer code.

So the idea is that we have formats marked by version 3, 3 + unstable, 4, 4 + unstable, etc

>I figured stable version was 3 and if we can do unstable versions, it would be 4

If we went with that suggestion, any RDGs written while 4 was an unstable storage format would be indistinguishable from RDGs written while 4 was stable at the json level, which would cause problems if the unstable format RDG doesn't match the stabilized format. Until something that expected the stable format fails dramatically, we would be unable to identify an RDG as unstable.
To get around this, we could say 4 was the unstable format and 5 was the stable format. But now we have to know that even versions are unstable and odd versions are stable or something like that. 

Hope that clarifies why adding a flag is much clearer than appending unstability to the existing storage_format_version.

Reply:
You make a good point.  The counterpoint is that any additional on-storage state is tough to manage, so even/odd versions are simpler.  But you've convinced me.

Reply:
Completely agree, we don't want to be throwing things in willy-nilly. My first go at implementing this was actually using even/odd versions but it felt obtuse :shrug: 
fingers crossed I don't regret this decision in the future